messaging/src/main/java/org/axonframework/eventhandling/pooled/Coordinator.java

```diff
@@ -295,12 +302,21 @@ private boolean initializeTokenStore() {
                 }
                 tokenStoreInitialized.set(true);
             } catch (Exception e) {
-                logger.info("Error while initializing the Token Store. This may simply indicate concurrent attempts to initialize.", e);
+                logger.info(
+                        "Error while initializing the Token Store. This may simply indicate concurrent attempts to initialize.",
+                        e);
             }
         });
         return tokenStoreInitialized.get();
     }
 
+    void resetRetryExponentialBackoff(int segmentId) {
```

Comment:
At first I was wondering why you made this package-private. But I see you pass this function to the builder of the `WorkPackage`, and as the `WorkPackage` is constructed by the `PooledStreamingEventProcessor`, you'd have too...

I have another suggestion. What if we add a package-private method to the `WorkPackage` (so, not a builder method) where things that construct the `WorkPackage` can set a `onbatchProcessed(Runnable)`?

Then, the `Coordinator` would, right after construction of the `WorkPackage`, invoke that method. In this style, I think we can make it a `Runnable` as well, as the `Coordinator` is perfectly aware of the `segmentId` it just created the `WorkPackage` for.

So, what do you think?

Reply:
At first, I wanted to keep it in the builder - to do not modify it after constructor from the outside of the class.
But... it sounds better, as you proposed to keep the Processor itself unaware of that this callback even exists. 

I've done it, you can check it now.